src/azure-cli-core/azure/cli/core/auth/credential_adaptor.py

```diff
@@ -27,16 +27,58 @@ def get_token(self, *scopes, **kwargs):
         """Get an access token from the main credential."""
         logger.debug("CredentialAdaptor.get_token: scopes=%r, kwargs=%r", scopes, kwargs)
 
-        # Discard unsupported kwargs: tenant_id, enable_cae
-        filtered_kwargs = {}
-        if 'data' in kwargs:
-            filtered_kwargs['data'] = kwargs['data']
+        msal_kwargs = _generate_msal_kwargs(kwargs)
+        return build_sdk_access_token(self._credential.acquire_token(list(scopes), **msal_kwargs))
+
+
+    def get_token_info(self, *scopes, options=None):
+        """Get an access token from the main credential."""
+        logger.debug("CredentialAdaptor.get_token_info: scopes=%r, options=%r", scopes, options)
+
+        msal_kwargs = _generate_msal_kwargs(options)
+        return _build_sdk_access_token_info(self._credential.acquire_token(list(scopes), **msal_kwargs))
 
-        return self._credential.get_token(*scopes, **filtered_kwargs)
 
     def get_auxiliary_tokens(self, *scopes, **kwargs):
         """Get access tokens from auxiliary credentials."""
         # To test cross-tenant authentication, see https://github.com/Azure/azure-cli/issues/16691
         if self._auxiliary_credentials:
-            return [cred.get_token(*scopes, **kwargs) for cred in self._auxiliary_credentials]
+            return [build_sdk_access_token(cred.acquire_token(list(scopes), **kwargs))
+                    for cred in self._auxiliary_credentials]
         return None
+
+
+def _generate_msal_kwargs(options=None):
```

Comment:
Unifying the `msal_kwargs` generation logic also fixes https://github.com/Azure/azure-cli/pull/30062#discussion_r1977122503.